### ISSUE TITLE
Is penaltymodel-lp 0.1.5 compatible with numpy 1.19.1 and scipy 1.5.2?

### ISSUE THREAD

**Issue by ccalaza** [dwavesystems/penaltymodel#134]
**Current Problem**
We are currently trying to install `dwave-ocean-sdk 4.2.0` in our HPC software stack at Forschungszentrum Juelich. Our environment has `Python 3.8.5`, `numpy 1.19.1` and `scipy 1.5.2`, but unfortunately the installation fails because `penaltymodel-lp 0.1.5` requires `numpy >=1.19.4` and `scipy >= 1.5.4`.

**Proposed Solution**
Would it be possible for you to test and confirm if `penaltymodel-lp 0.1.5` works correctly with `numpy 1.19.1` and `scipy 1.5.2`?

Many thanks in advance!

**Comment by arcondello** [comment]
Hi @ccalaza, looking into it now, will get back to you shortly.

**Comment by arcondello** [comment]
It appears that those versions do work. I have made a PR to slightly lower the minimum bound, see https://github.com/dwavesystems/penaltymodel/pull/135.

Do you need a release of `penaltymodel-lp` and/or the sdk? Or does knowing that it will work suffice?

**Comment by ccalaza** [comment]
I just needed confirmation, I'll patch the installation myself. Once again thanks a lot for the super-quick reaction!

**Comment by arcondello** [comment]
Hi @ccalaza , following up on this. We'd like to raise our minimum supported SciPy version to 1.6.0 to take advantage of the `'highs'` linear programming method, see [link](https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.linprog.html#scipy.optimize.linprog).

I am curious what the impact on y'all would be? How much of a blocker would this be to using Ocean?

**Comment by arcondello** [comment]
I was able to find a workaround (see #138 ), but still interested to know for future reference.

**Comment by ccalaza** [comment]
Hi Alex,

Happy new year and thanks for checking with us, that's very much appreciated!

Unfortunately, as of now we are stuck with Scipy 1.5.2. But the plan is to upgrade to a new environment with Scipy 1. 7.1 in the coming months, hopefully by the beginning of Q2.

X
Carlos

**Comment by arcondello** [comment]
Hi @ccalaza , checking in again! Any chance it's been updated?

**Comment by ccalaza** [comment]
Hi Alex! 

We are currently almost done switching to our software stage "2022" with NumPy 1.21.3 and SciPy 1.7.1. If everything goes according to plan we will stick to these versions at least until around this time next year, when we switch to stage "2023".

Unless of course there's some bug in these packages or their dependencies which forces us to upgrade before then, or for some unlikely reason we need to deploy a "2022b" stage in the 2nd half of 2022. Even then we may not need to upgrade NumPy and Scipy, though.

TLDR, we are extremely likely to stick to NumPy 1.21.3 and SciPy 1.7.1 until at least Q2 2023.

X
Carlos
